src/libmw/include/mw/mmr/MMRInfo.h

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <mw/models/crypto/Hash.h>
+#include <boost/optional.hpp>
+
+/// <summary>
+/// Represents the state of the MMR files with the matching index.
+/// </summary>
+struct MMRInfo : public Traits::ISerializable
+{
+    MMRInfo()
+        : index(0), pruned(mw::Hash()), compact_index(0), compacted(boost::none) { }
+    MMRInfo(uint32_t index_in, mw::Hash pruned_in, uint32_t compact_index_in, boost::optional<mw::Hash> compacted_in)
+        : index(index_in), pruned(std::move(pruned_in)), compact_index(compact_index_in), compacted(std::move(compacted_in)) { }
+
+    // File number of the MMR files.
+    uint32_t index;
+
+    // Hash of latest header this MMR represents.
+    mw::Hash pruned;
+
+    // File number of the PruneList bitset.
+    uint32_t compact_index;
+
+    // Hash of the header this MMR was compacted for.
+    // You cannot rewind beyond this point.
+    boost::optional<mw::Hash> compacted;
+
+    Serializer& Serialize(Serializer& serializer) const noexcept final
+    {
+        return serializer
+            .Append<uint32_t>(index)
+            .Append(pruned)
+            .Append(compacted.value_or(mw::Hash()));
+    }
+
+    static MMRInfo Deserialize(Deserializer& deserializer)
+    {
+        uint32_t index = deserializer.Read<uint32_t>();
+        mw::Hash pruned = deserializer.Read<mw::Hash>();
+        uint32_t compact_index = deserializer.Read<uint32_t>();
```

Comment:
Deserialize without corresponding serialize.

Reply:
I ran into that issue when testing a few days ago. I'll be adding the fix and some unit-tests for this with the next commit.